contracts/feature-tests/use-module/tests/gov_module_whitebox_test.rs

```diff
@@ -0,0 +1,603 @@
+        |sc| {
+            sc.execute(proposal_id);
+        },
+        |r| {
+            r.assert_user_error("Can only execute queued proposals");
+        },
+    );
+
+    // try queue before voting ends
+    world.whitebox_call_check(
+        &use_module_whitebox,
+        ScCallStep::new().from(FIRST_USER_ADDRESS_EXPR).no_expect(),
+        |sc| {
+            sc.queue(proposal_id);
+        },
+        |r| {
+            r.assert_user_error("Can only queue succeeded proposals");
+        },
+    );
+
+    current_block_nonce += VOTING_PERIOD_BLOCKS;
+    world.set_state_step(SetStateStep::new().block_nonce(current_block_nonce));
+
+    // try queue not enough votes
+    world.whitebox_call_check(
+        &use_module_whitebox,
+        ScCallStep::new().from(FIRST_USER_ADDRESS_EXPR).no_expect(),
+        |sc| {
+            sc.queue(proposal_id);
+        },
+        |r| {
+            r.assert_user_error("Can only queue succeeded proposals");
+        },
+    );
+
+    // user 1 vote again
+    current_block_nonce = 20;
+    world.set_state_step(SetStateStep::new().block_nonce(current_block_nonce));
+    world.whitebox_call(
+        &use_module_whitebox,
+        ScCallStep::new()
+            .from(FIRST_USER_ADDRESS_EXPR)
+            .esdt_transfer(GOV_TOKEN_ID, 0, rust_biguint!(200)),
+        |sc| {
+            sc.vote(proposal_id, VoteType::UpVote);
+        },
+    );
+
+    // owner downvote
+    world.whitebox_call(
+        &use_module_whitebox,
+        ScCallStep::new().from(OWNER_ADDRESS_EXPR).esdt_transfer(
+            GOV_TOKEN_ID,
+            0,
+            rust_biguint!(200),
+        ),
+        |sc| {
+            sc.vote(proposal_id, VoteType::DownVote);
+        },
+    );
+
+    // try queue too many downvotes
+    current_block_nonce = 45;
+    world.set_state_step(SetStateStep::new().block_nonce(current_block_nonce));
+    world.whitebox_call_check(
+        &use_module_whitebox,
+        ScCallStep::new().from(FIRST_USER_ADDRESS_EXPR).no_expect(),
+        |sc| {
+            sc.queue(proposal_id);
+        },
+        |r| {
+            r.assert_user_error("Can only queue succeeded proposals");
+        },
+    );
+
+    // user 1 vote again
+    current_block_nonce = 20;
+    world.set_state_step(SetStateStep::new().block_nonce(current_block_nonce));
+    world.whitebox_call_check(
+        &use_module_whitebox,
+        ScCallStep::new()
+            .from(FIRST_USER_ADDRESS_EXPR)
+            .esdt_transfer(GOV_TOKEN_ID, 0, rust_biguint!(200))
+            .no_expect(),
+        |sc| {
+            sc.vote(proposal_id, VoteType::UpVote);
+        },
+        |r| {
+            r.assert_user_error("Already voted for this proposal");
+        },
+    );
+
+    // user 3 vote again
+    world.whitebox_call(
+        &use_module_whitebox,
+        ScCallStep::new()
+            .from(THIRD_USER_ADDRESS_EXPR)
+            .esdt_transfer(GOV_TOKEN_ID, 0, rust_biguint!(200)),
```

Comment:
I think it should work without the rust_biguint! we should have direct conversion from small int